hack/common.sh

```diff
@@ -862,4 +865,101 @@ function os::build::find-binary() {
   local path=$( (ls -t $(os::build::get-bin-output-path "${os_root}")/${bin}) 2>/dev/null || true | head -1 )
   echo "$path"
 }
-readonly -f os::build::find-binary
\ No newline at end of file
+readonly -f os::build::find-binary
+
+# os::build::environment::create creates a docker container with the default variables.
+# arguments are passed directly to the container, OS_BUILD_ENV_GOLANG, OS_BUILD_ENV_IMAGE,
+# and OS_RELEASE_DOCKER_ARGS can be used to customize the container. The docker socket
+# is mounted by default and the output of the command is the container id.
+os::build::environment::create() {
+  golang_version="${OS_BUILD_ENV_GOLANG}"
+  release_image="${OS_BUILD_ENV_IMAGE}"
+  additional_context="${OS_BUILD_ENV_DOCKER_ARGS:-}"
+  if [[ -z "${additional_context}" && "${OS_BUILD_ENV_USE_DOCKER:-y}" == "y" ]]; then
+    additional_context="--privileged -v /var/run/docker.sock:/var/run/docker.sock"
+
+    if [[ "${OS_BUILD_ENV_LOCAL_DOCKER:-n}" == "y" ]]; then
+      # if OS_BUILD_ENV_LOCAL_DOCKER==y, add the local OS_ROOT as the bind mount to the working dir
+      # and set the running user to the current user
+      workingdir=$( os::build::environment::release::workingdir )
```

Comment:
Quote subshells to avoid word expansion
```sh
local workingdir #not inline so we can't mask the return code of the subshell
workingdir="$( ... )"
```